src/openforms/registrations/contrib/objects_api/tests/test_backend_v2.py

```diff
@@ -633,6 +645,106 @@ def test_object_ownership_not_validated_if_new_object(self):
         except AssertionError:
             self.fail("Assertion should have passed.")
 
+    @patch(
+        "openforms.contrib.haal_centraal.clients.HaalCentraalConfig.get_solo",
+        return_value=HaalCentraalConfig(
+            brp_personen_service=ServiceFactory.build(
+                api_root="http://localhost:5010/haalcentraal/api/brp/"
+            ),
+            brp_personen_version=BRPVersions.v20,
+        ),
+    )
+    @patch(
+        "openforms.config.models.GlobalConfiguration.get_solo",
+        return_value=GlobalConfiguration(
+            family_members_data_api=FamilyMembersDataAPIChoices.haal_centraal
+        ),
+    )
+    def test_submission_with_partners_component(self, m, n):
+        submission = SubmissionFactory.from_components(
+            [
+                {
+                    "key": "partners",
+                    "type": "partners",
+                    "registration": {
+                        "attribute": RegistrationAttribute.partners,
+                    },
+                }
+            ],
+            auth_info__value="000009921",
+            auth_info__attribute=AuthAttribute.bsn,
+            completed_on=datetime(2024, 11, 9, 15, 30, 0).replace(tzinfo=UTC),
+        )
+        FormVariableFactory.create(
+            key="partners_immutable",
+            form=submission.form,
+            user_defined=True,
+            prefill_plugin=FM_PLUGIN_IDENTIFIER,
+            prefill_options={
+                "type": "partners",
+                "mutable_data_form_variable": "partners",
+                "min_age": None,
+                "max_age": None,
+            },
+        )
+
+        serializer = ObjectsAPIOptionsSerializer(
+            data={
+                "version": 2,
+                "objects_api_group": self.objects_api_group.pk,
+                # See the docker compose fixtures for more info on these values:
+                "objecttype": UUID("59cdc902-576b-495f-ae63-c9c78b8afc09"),
+                "objecttype_version": 1,
+                "upload_submission_csv": False,
+                "update_existing_object": False,
+                "variables_mapping": [
+                    {
+                        "variable_key": "partners",
+                        "target_path": ["partners"],
+                    },
+                ],
+                "transform_to_list": [],
+                "iot_attachment": "",
+                "iot_submission_csv": "",
+                "iot_submission_report": "",
+            }
+        )
+
+        assert serializer.is_valid()
+
+        v2_options: RegistrationOptionsV2 = serializer.validated_data
```

Comment:
is there a particular reason why you don't instantiate the `v2_options` directly without the serializer?

Reply:
okay - I think a lot of these tests are from before we did strict(er) type checking and are due some restructuring to remove the unneeded stuff, but it's not a limitation for this PR